ember-client/shared/src/main/scala/org/http4s/ember/client/internal/WebSocketHelpers.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2019 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.ember.client.internal
+
+import cats.MonadThrow
+import cats.syntax.all._
+import org.http4s._
+import org.http4s.crypto.Hash
+import org.http4s.crypto.HashAlgorithm
+import org.http4s.headers.Connection
+import org.http4s.headers._
+import org.http4s.syntax.all._
+import org.http4s.websocket.Rfc6455
+import org.typelevel.ci._
+import scodec.bits.ByteVector
+
+private[internal] object WebSocketHelpers {
+
+  private[this] val webSocketProtocol = Protocol(ci"websocket", None)
+
+  /** Validate the opening handshake response from the server
+    * https://datatracker.ietf.org/doc/html/rfc6455#page-6
+    */
+  def validateServerHandshake[F[_]](
+      response: Response[F],
+      secWebSocketKey: String,
+  )(implicit F: MonadThrow[F]): F[Boolean] =
```

Comment:
This looks great! Can we add a `WebSocketHelpersSuite` that validates a successful response, and also one response for each of the different failure kinds?